### PR TITLE
Add regression issue template

### PR DESCRIPTION
**Pull request by camelid**
Feel free to suggest improvements!

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/regression.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+---
+name: Regression
+about: Report a regression in Rust.
```
Comment (jyn514) [inline .github/ISSUE_TEMPLATE/regression.md:3]:
```suggestion
about: Report a regression the Rust compiler or `rustdoc`.
```

File: .github/ISSUE_TEMPLATE/regression.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+```
+<backtrace>
+```
+
+</p>
+</details>
+
+<!--
+Please change `nightly` to the release channel you found this regression in.
```
Comment (jyn514) [inline .github/ISSUE_TEMPLATE/regression.md:60]:
```suggestion
Please change `nightly` to the release channel you found this regression in (stable, beta, or nightly).
```

File: .github/ISSUE_TEMPLATE/regression.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+-->
+
+### Code
+
+I tried this code:
+
+```rust
+<code>
+```
+
```
Comment (jyn514) [inline .github/ISSUE_TEMPLATE/regression.md:18]:
```suggestion
I expected to see this happen: *explanation*

Instead, this happened: *explanation*

```

File: .github/ISSUE_TEMPLATE/regression.md (hunk 1 above)
Comment (bugadani) [reply .github/ISSUE_TEMPLATE/regression.md:3]: I think there's something missing in this suggestion

Comment (camelid) [reply .github/ISSUE_TEMPLATE/regression.md:3]: I just went off of what was there for "Bug Report":

image

Should it be changed?

Comment (jyn514) [reply .github/ISSUE_TEMPLATE/regression.md:3]: Oh hmm if that's what it is on the other issue template no need to change it here.

File: .github/ISSUE_TEMPLATE/regression.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+</p>
+</details>
+
+<!--
+Please change `nightly` to the release channel you found this regression in (stable, beta, or nightly).
+-->
+
+@rustbot modify labels: regression-from-stable-to-nightly
```
Comment (Mark-Simulacrum) [inline .github/ISSUE_TEMPLATE/regression.md:66]: Hm, I wonder if this will turn out to be a bad idea. Maybe we can have the default be something like "regression-untriaged" (I can create the label).

I worry that we'll otherwise get people filing nightly regressions, which are actually beta regressions.

Comment (camelid) [reply .github/ISSUE_TEMPLATE/regression.md:66]: > Maybe we can have the default be something like "regression-untriaged" (I can create the label).

That seems like a good idea 👍 

> I worry that we'll otherwise get people filing nightly regressions, which are actually beta regressions.

Well it would probably be discovered when it's triaged, but yeah it probably makes sense to default to `regression-untriaged` so there's no confusion.

Comment (camelid) [reply .github/ISSUE_TEMPLATE/regression.md:66]: Maybe we have the default be `regression-untriaged` and then say in a comment that you can change it if you know where it occurred?

Comment (Mark-Simulacrum) [reply .github/ISSUE_TEMPLATE/regression.md:66]: Yes, that seems reasonable. If it really becomes a problem we'll just remove the second bit.

File: .github/ISSUE_TEMPLATE/regression.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+<backtrace>
+```
+
+</p>
+</details>
+
+<!--
+If you know when this regression occurred, please add a line like below, replacing `{channel}` with one of stable, beta, or nightly.
+
+@rustbot modify labels: regression-from-stable-to-{channel}
```
Comment (camelid) [inline .github/ISSUE_TEMPLATE/regression.md:66]: Hmm, should I change it to this so that the `regression-untriaged` label isn't added in addition to `regression-from-stable-to-{channel}`?

```suggestion
@rustbot modify labels: +regression-from-stable-to-{channel} -regression-untriaged
```

Comment (Mark-Simulacrum) [reply .github/ISSUE_TEMPLATE/regression.md:66]: That seems good, yes.

File: .github/ISSUE_TEMPLATE/regression.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
+---
+name: Regression
+about: Report a regression in Rust.
+labels: C-bug regression-untriaged
+---
+<!--
+Thank you for filing a regression report! 🐛 Please provide a short summary of the regression,
+along with any information you feel is relevant to replace it.
```
Comment (jyn514) [inline .github/ISSUE_TEMPLATE/regression.md:8]: Did you mean 'replicate'?
```suggestion
along with any information you feel is relevant to replicate it.
```

File: .github/ISSUE_TEMPLATE/regression.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+I expected to see this happen: *explanation*
+
+Instead, this happened: *explanation*
+
+### Version it worked on
+
+<!--
+Provide the most recent version this worked on, for example:
+
+It most recently worked on: Rust 1.47
```
Comment (jyn514) [inline .github/ISSUE_TEMPLATE/regression.md:28]: I don't think we need both this and the text below. Maybe put 'Rust 1.47' in the `version` comment?

File: .github/ISSUE_TEMPLATE/regression.md (hunk 6 above)
Comment (camelid) [reply .github/ISSUE_TEMPLATE/regression.md:8]: Oops, yes :)

File: .github/ISSUE_TEMPLATE/regression.md (hunk 7 above)
Comment (camelid) [reply .github/ISSUE_TEMPLATE/regression.md:28]: Hmm, I think it would be confusing that way. Personally I would prefer to be a bit verbose rather than potentially unclear.

Comment (jyn514) [reply .github/ISSUE_TEMPLATE/regression.md:28]: Ok, seems fine.

**Comment by camelid** [comment]
Squashed. Ready for review!

**Comment by jyn514** [comment]
We should wait until https://github.com/rust-lang/rust/pull/77769 lands to merge this IMO.

**Comment by camelid** [comment]
Yes, I should have mentioned that.

**Comment by LeSeulArtichaut** [comment]
I'm worried that this template sort of intersects with both the Bug and ICE ones. How could we make it easier for a new bug reporter which template to use?

File: .github/ISSUE_TEMPLATE/regression.md (hunk 4 above)
Comment (spastorino) [reply .github/ISSUE_TEMPLATE/regression.md]: By the way, I've noticed as @Mark-Simulacrum points out that from time to time we've things tagged as nightly regressions that are beta regressions. As part of @rust-lang/wg-prioritization effort, I've been checking the ones that we track for the compiler weekly meeting.

I wonder if we could make a process on Triage WG and/or Prioritization WG and even automate some stuff to more actively check that the regressions are correctly labelled.

**Comment by LeSeulArtichaut** [comment]
#77769 landed

**Comment by camelid** [comment]
> I'm worried that this template sort of intersects with both the Bug and ICE ones. How could we make it easier for a new bug reporter which template to use?

Any ideas? :P

**Comment by jyn514** [comment]
Maybe change 'Bug' -> 'Existing Bug'? 'existing' is a little ambiguous if you don't know it's talking about releases, though ...

**Comment by camelid** [comment]
Another issue is that people might not know what a "regression" is :/

File: .github/ISSUE_TEMPLATE/regression.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
+---
+name: Regression
+about: Report a regression in Rust.
```
Comment (LeSeulArtichaut) [inline .github/ISSUE_TEMPLATE/regression.md:3]: What about
```suggestion
about: Report a bug for something that used to work but doesn't anymore.
```

**Comment by LeSeulArtichaut** [comment]
> Maybe change 'Bug' -> 'Existing Bug'? 'existing' is a little ambiguous if you don't know it's talking about releases, though ...

I personally find it more confusing

File: .github/ISSUE_TEMPLATE/regression.md (hunk 8 above)
Comment (camelid) [reply .github/ISSUE_TEMPLATE/regression.md:3]: That seems good. What do other people think?

**Comment by mark-i-m** [comment]
How about something like "Breakage on Stable Rust"

**Comment by apiraino** [comment]
> How about something like "Breakage on Stable Rust"

If I'm not wrong a regression in this repository can be:
* from stable to stable (i.e. a stable breaks another stable - perhaps the case that you mention)
* from stable to beta or nightly (i.e. a nightly/beta breaks stable)
* from beta to nightly (i.e. a nightly breaks a beta)

My suggestion "The issue breaks a previous Rust version"?

File: .github/ISSUE_TEMPLATE/regression.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+labels: C-bug regression-untriaged
+---
+<!--
+Thank you for filing a regression report! 🐛 Please provide a short summary of the regression,
+along with any information you feel is relevant to replicate it.
```
Comment (apiraino) [inline .github/ISSUE_TEMPLATE/regression.md:8]: I like what LeSeulArtichaut suggested to explain what a regression is. Therefore I'd clarify a little bit here:

```suggestion
Thank you for filing a regression! 🐛  A regression is an error in a component that used to work but doesn't anymore.

Please provide a short summary of the regression, along with any information you feel is relevant to replicate it.
```

File: .github/ISSUE_TEMPLATE/regression.md (hunk 8 above)
Comment (apiraino) [reply .github/ISSUE_TEMPLATE/regression.md:3]: left a [comment](https://github.com/rust-lang/rust/pull/77725#issuecomment-706756620) with a possible alternative

File: .github/ISSUE_TEMPLATE/regression.md (hunk 9 above)
Comment (camelid) [reply .github/ISSUE_TEMPLATE/regression.md:8]: Hmm, not *all* regressions are new errors that didn't used to occur. What do you think of this?

```suggestion
Thank you for filing a regression report! 🐛 A regression is something that changed between versions of Rust but was not supposed to.

Please provide a short summary of the regression,
along with any information you feel is relevant to replicate it.
```

File: .github/ISSUE_TEMPLATE/regression.md (hunk 8 above)
Comment (camelid) [reply .github/ISSUE_TEMPLATE/regression.md:3]: Actually just realized that not all regressions are things that used to work but don't anymore: they can also be things that shouldn't work but do. Maybe this?

```suggestion
about: Report something that unexpectedly changed between Rust versions.
```

Cc @Mark-Simulacrum -- what do you think?

**Comment by apiraino** [comment]
For NextCloud users passing by, this issue breaks the NextCloud installation because PHP 8.0.0RC1 is pulled and not yet supported by NextCloud.

File: .github/ISSUE_TEMPLATE/regression.md (hunk 9 above)
Comment (apiraino) [reply .github/ISSUE_TEMPLATE/regression.md:8]: Probably both more correct than "my" definition and clearer in layman's terms. :+1:

**Comment by Mark-Simulacrum** [comment]
Please squash the commits; r=me after that. I imagine we'll want to iterate but seems good to get something merged in the meantime.

**Comment by camelid** [comment]
Squashed!
